tests/storage/memorydump.go

```diff
@@ -610,4 +613,97 @@ var _ = SIGDescribe("Memory dump", func() {
+		)
+
+		memoryDumpVirtctlDownload := func(name, namespace, outputFile string) {
+			By("Invoking virtlctl memory dump download")
+			commandAndArgs := []string{commandMemoryDump, "download", name, virtCtlNamespace, namespace}
+			commandAndArgs = append(commandAndArgs, fmt.Sprintf(virtCtlOutputFile, outputFile))
+			memorydumpCommand := clientcmd.NewRepeatableVirtctlCommand(commandAndArgs...)
+			Eventually(func() error {
+				return memorydumpCommand()
+			}, 6*time.Second, 1*time.Second).ShouldNot(HaveOccurred())
+		}
+
+		memoryDumpVirtctlGetWithDownload := func(name, namespace, claimName, outputFile string) {
+			By("Invoking virtlctl memory dump")
+			commandAndArgs := []string{commandMemoryDump, "get", name, virtCtlNamespace, namespace}
+			commandAndArgs = append(commandAndArgs, fmt.Sprintf(virtCtlClaimName, claimName))
+			commandAndArgs = append(commandAndArgs, fmt.Sprintf(virtCtlOutputFile, outputFile))
+			memorydumpCommand := clientcmd.NewRepeatableVirtctlCommand(commandAndArgs...)
+			Eventually(func() error {
+				return memorydumpCommand()
+			}, 6*time.Second, 1*time.Second).ShouldNot(HaveOccurred())
+		}
+
+		memoryDumpVirtctlCreateWithDownload := func(name, namespace, claimName, outputFile string) {
+			By("Invoking virtlctl memory dump with create flag")
+			commandAndArgs := []string{commandMemoryDump, "get", name, virtCtlNamespace, namespace}
+			commandAndArgs = append(commandAndArgs, fmt.Sprintf(virtCtlClaimName, claimName))
+			commandAndArgs = append(commandAndArgs, virtCtlCreate)
+			commandAndArgs = append(commandAndArgs, fmt.Sprintf(virtCtlOutputFile, outputFile))
+			memorydumpCommand := clientcmd.NewRepeatableVirtctlCommand(commandAndArgs...)
+			Eventually(func() error {
+				err := memorydumpCommand()
+				if err != nil {
+					_, getErr := virtClient.CoreV1().PersistentVolumeClaims(namespace).Get(context.Background(), claimName, metav1.GetOptions{})
+					if getErr == nil {
+						// already created the pvc can't call the memory dump command with
+						// create-claim flag again
+						By("Error memory dump command after claim created")
+						memoryDumpVirtctlGetWithDownload(name, namespace, claimName, outputFile)
+						return nil
+					}
+				}
+				return err
+			}, 6*time.Second, 1*time.Second).ShouldNot(HaveOccurred())
+		}
+
+		BeforeEach(func() {
+			if !checks.IsOpenShift() {
+				Skip("Not on openshift")
+			}
+			sc, exists := libstorage.GetRWOFileSystemStorageClass()
+			if !exists {
+				Skip("Skip no filesystem storage class available")
+			}
+			libstorage.CheckNoProvisionerStorageClassPVs(sc, numPVs)
+
+			vm = createAndStartVM()
+		})
+
+		AfterEach(func() {
+			if vm != nil {
+				deleteVirtualMachine(vm)
+			}
+			pvc, err := virtClient.CoreV1().PersistentVolumeClaims(util.NamespaceTestDefault).Get(context.Background(), memoryDumpPVCName, metav1.GetOptions{})
+			if err == nil && pvc != nil {
+				deletePVC(pvc)
+			}
+
+			_, err = os.Stat(outputFile)
```

Comment:
Why do we need to check here the existence of the outputFile. Is this part of the cleanup or the test?